addons/sale_loyalty/models/sale_order.py

```diff
@@ -913,8 +913,16 @@ def _program_check_compute_points(self, programs):
                         points += rule.reward_point_amount
                     elif rule.reward_point_mode == 'money':
                         # Compute amount paid for rule
-                        # NOTE: this does not account for discounts -> 1 point per $ * (100$ - 30%) will result in 100 points
-                        amount_paid = sum(max(0, line.price_total) for line in order_lines if line.product_id in rule_products)
+                        # NOTE: this accounts for discounts -> 1 point per $ * (100$ - 30%) will result in 70 points
+                        lines_for_points = []
+                        for line in self.order_line - self._get_no_effect_on_threshold_lines():
```

Comment:
With my previous comment I explain that it can't be nested out of the program loop :smile: However it cannot be nested out of the rule loop if I apply Valentin's comment. So the question is do I remove the loop here 
```python
amount_paid = sum(line.price_total for line in lines_for_points if line.product_id in so_products_per_rule.get(rule, []))
``` 
This even remove the need to remember the `lines_for_points` or do I leave it and un-nest the original loop where I keep `lines_for_points` ?

Reply:
It depends how many programs are filtered out by the condition in your loop. If there are not too many, then VCR's suggestion is a bit better IMO because it simplifies the code (and keeps the logic in one single place). And I suppose there are indeed not that many, as most people probably won't be using loads of Gift Cards and eWallets on a single order.

Would be nice to have some stats about this but it's not worth the effort IMO :sweat_smile:

@chevalierv WDYT?

Reply:
Indeed, I guess that it's unlikely to have many programs here